Proofs/SHA512/SHA512_block_armv8_rules.lean

```diff
@@ -134,17 +134,25 @@ theorem concat_of_rsh_is_msb_128 (x y : BitVec 64) :
   (x ++ y) >>> 64 = BitVec.zeroExtend 128 x := by
   bv_check "lrat_files/Sha512_block_armv8_rules.lean-concat_of_rsh_is_msb_128-101-2.lrat"
 
--- (FIXME) Generalize to arbitrary-length bitvecs.
 theorem truncate_of_concat_is_lsb_64 (x y : BitVec 64) :
   BitVec.zeroExtend 64 (x ++ y) = y := by
   bv_check "lrat_files/Sha512_block_armv8_rules.lean-truncate_of_concat_is_lsb_64-106-2.lrat"
 
--- (FIXME) Generalize to arbitrary-length bitvecs.
+theorem truncate_of_concat_is_lsb (w : Nat) (x y : BitVec w) :
+  BitVec.zeroExtend w (x ++ y) = y := by
+  ext
+  simp [BitVec.leftshift_n_or_mod_2n]
```

Comment:
Thanks for this! In LNSym, we prefer using `simp only` wherever possible, so could you please fix that? Note that you can also use `BitVec.truncate_append` for a smaller/cleaner proof here.

Also, I acknowledge that this file has badly named lemmas (it precedes Lean's BitVec lemma dev) --- this lemma should've really used `zeroExtend` and `append` in the name.